src/components/Database/DataViewer/ImportDialog.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Dialog,
+  DialogActions,
+  DialogButton,
+  DialogContent,
+  DialogTitle,
+} from '@rmwc/dialog';
+import * as React from 'react';
+import { useState } from 'react';
+
+import { Callout } from '../../common/Callout';
+import { Field } from '../../common/Field';
+import { FileField } from '../../common/FileField';
+import { Spinner } from '../../common/Spinner';
+import { DatabaseApi } from '../api';
+import { useApi } from '../ApiContext';
+import styles from './ImportDialog.module.scss';
+
+export interface Props {
+  api: DatabaseApi;
+  reference: firebase.database.Reference;
+  onComplete: (reference?: firebase.database.Reference, file?: File) => void;
+}
+
+export const ImportDialog: React.FC<Props> = ({ reference, onComplete }) => {
+  const api = useApi();
```

Comment:
Please remove this and the `useApi()` hook + Context. This component already takes `api` in the props, and there is no reason to introduce a Context. Also general comment: I'd strongly prefer passing a prop if a prop will do, since it is straightforward to reason about, statically type-checked, and does not throw runtime errors.

Reply:
I thought it might be needed "lower down" later... but I can remove for now.

Reply:
Yeah, my advice would be keep "drilling" the property until at some point we decide it's too annoying, and migrate to Context. We followed the route for Firestore viewer and now it's in Context-land, but I wouldn't recommend introducing a Context "just in case" before drilling drags us down to much. It's easy to migrate drilling to Context but the opposite is much harder due to the dynamic nature.